lib/rails_state_machine/event.rb

```diff
@@ -64,8 +71,23 @@ def allowed_from?(state_name)
       @transitions_by_state_name.key?(state_name&.to_sym)
     end
 
-    def future_state_name(state_name)
+    def future_state_name(state_name, raise_on_invalid)
       find_transition_from(state_name).to
+    rescue TransitionNotFoundError
+      if raise_on_invalid
+        raise
```

Comment:
Calling raise within a rescue block is reraising the causing exception.

> With no arguments, raises the exception in $! or raises a [RuntimeError](https://ruby-doc.org/3.2.2/RuntimeError.html) if $! is nil.

```
def foo
  raise 'hello'
rescue
  puts $!.inspect
  raise
end

foo
#<RuntimeError: hello>
(irb):2:in `foo': hello (RuntimeError)
```

So this might be not necessary, or did I misunderstand the suggestion?

Reply:
Thanks, I did not remember that :+1: